pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -364,6 +369,24 @@ pub mod pallet {
 			.map_err(Into::into)
 		}
 
+		/// Transfer an asset from an account to another one
+		pub fn transfer(
+			asset_id: AssetId,
+			from: T::AccountId,
+			to: T::AccountId,
+			amount: U256,
+		) -> Result<(), evm::EvmError> {
+			frame_support::storage::with_storage_layer(|| {
```

Comment:
We can remove the `with_storage_layer`, https://github.com/moonbeam-foundation/moonbeam/pull/3258 changes the mechanism to properly handle this.